astropy/io/ascii/core.py

```diff
@@ -438,9 +438,9 @@ def __call__(self, lines):
         lines : list
             List of table lines
 
-        Returns
-        -------
-        lines : iterator
+        Yields
+        ------
+        lines : list
```

Comment:
Like Returns, Yields is meant to describe the value yielded by the generator at each step of the generator. Reading https://docs.python.org/3.8/library/csv.html?highlight=csv#csv.reader, this is a "list of str". Since the option ``process_val`` can modify the value, in principle into any type, I suggest 

```
        Yields
        ------
        line : list of (str or object)
```

Reply:
✅

Reply:
> Like Returns, Yields is meant to describe the value yielded by the generator at each step of the generator. Reading https://docs.python.org/3.8/library/csv.html?highlight=csv#csv.reader, this is a "list of str". Since the option `process_val` can modify the value, in principle into any type, I suggest
> 

In that option it calls `BaseSplitter.process_val`, which just returns `val.strip()`, so the list items will still be strings (to be passed to the parser later in the process).
I tend to follow the description in the top docstring "each iterator output is a list of the split line values." e.g.
```
        values : list of str
            Each line's split values
```

Reply:
Great. Implementing the example. It's good as-is.